### DISCUSSION TITLE
GitHub package retention policy support

### DISCUSSION THREAD

**Discussion by taehwanno** [community/community#38962] [Packages]
### Select Topic Area

Product Feedback

#### Question

Is there a way to set the GitHub package's retention policy like below?
- Remove if 90 days have passed since the version was published
- Remove if not downloaded within 90 days

#### Context

we have published some codes to github private package versioning with build number (e.g. commit sha). but the github package max version count limit is 1000 in based on [limits for published npm versions](https://docs.github.com/en/enterprise-server@3.6/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#limits-for-published-npm-versions). 

> If you publish over 1,000 npm package versions to GitHub Packages, you may see performance issues and timeouts occur during usage.
>
> In the future, to improve performance of the service, you won't be able to publish more than 1,000 versions of a package on GitHub. Any versions published before hitting this limit will still be readable.

We can use actions like [this](https://github.com/marketplace/actions/delete-package-versions), but I think the ideal direction is to support a retention policy that cleans up packages that have not been used in the last few days. In [RESTful API](https://docs.github.com/en/rest/packages#get-a-package-version-for-an-organization) related to package, there is no API that returns the date of the last download.

**Comment by handipriyono** [comment]
same question, 
btw have you received the answer or solved this?/

**Comment by ashah-carsales** [comment]
Any updates, having the same issue

**Comment by i-fedichkin-upstars** [comment]
Same here, we will have the same issue with this limit
